src/pages/Login/Login.tsx

```diff
@@ -0,0 +1,145 @@
+import { setLoginStatus } from '../../store/reducers/AuthSlice'
+import { IErrorResponse } from '../../models/IErrorResponse';
+import { ISigInParams } from '../../models/ISigInParams';
+import { IAlertTypeProps, showAlert } from '../../store/reducers/AlertSlice';
+
+const TextField = lazy(() => import(/* webpackChunkName: "TextField" */ '../../components/TextField/TextField'))
+const Loader = lazy(() => import(/* webpackChunkName: "Loader" */ '../../components/Loader/Loader'))
+
+const schema = yup.object()
+  .shape({
+    password: yup.string()
+      .required('Пароль не указан.')
+      .min(8, 'Пароль может содержать только латинские буквы.')
+      .matches(/[a-zA-Z]/, 'Пароль может содержать только латинские буквы.'),
+    login: yup.string()
+      .required('Логин не указан.')
+  })
+
+const Login = () => {
+  const methods = useForm<ISigInParams>({
+    defaultValues: {
+      [InputName.login]: 'Test0010',
+      [InputName.password]: 'Abrikosov8436259'
+    },
+    mode: 'onBlur',
+    resolver: yupResolver(schema)
+  });
+
+  const [fetchLogin, {
+    isLoading,
+    data,
+    error,
+    isSuccess,
+    isError
+  }] = useFetchSigInMutation();
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const [isShowPassword, setIsShowPassword] = useState(false);
+
+  const onSubmit = (value: ISigInParams) => fetchLogin(value)
+
+  useEffect(() => {
+    if (isSuccess) {
+      dispatch(setLoginStatus(true))
+      navigate(RouterLinks.HOME)
+    }
+  }, [data])
+
+  useEffect(() => {
+    if (isError) {
+      const err = ((error) as IErrorResponse);
+      dispatch(showAlert({
+        text: err?.data?.reason ?? MESSAGES_TEXT.ERROR_OCCURRED,
+        type: TYPES_ALERT.ERROR as IAlertTypeProps
+      }));
+    }
+  }, [error])
+
+  return (
+    <FormProvider {...methods}>
+      <form onSubmit={methods.handleSubmit(onSubmit)} className={styles.form}>
+        {
+          isLoading && <Loader />
+        }
+        <Grid
+          container
+          spacing={0}
+          direction='column'
+          alignItems='center'
+          justifyContent='center'
+          className={styles.layout}
+        >
+          <Grid item xs={12} className={styles.input}>
+            <TextField name={InputName.login} label={InputLabel.login} autoFocus />
+          </Grid>
+
+          <Grid item xs={12} className={styles.input}>
+            <TextField
+              type={isShowPassword ? '' : InputName.password}
+              name={InputName.password}
+              label={InputLabel.password}
+            />
+          </Grid>
+
+          <Grid
+            container
+            alignItems='center'
+            justifyContent='center'
+          >
+            <Grid item>
+              <FormControlLabel
+                control={(
+                  <Checkbox
+                    color='primary'
+                    onClick={() => setIsShowPassword((pre) => !pre)}
```

Comment:
🧐
Можно создать мемоизированный `togglePasswordVisiblity` метод и использовать его вместо инлайн функции
```suggestion
                    onClick={togglePasswordVisiblity}
```

Reply:
Готово!